src/test/ui/type-alias-impl-trait/generic_duplicate_param_use6.stderr

```diff
@@ -1,19 +1,14 @@
 error: concrete type differs from previous defining opaque type use
   --> $DIR/generic_duplicate_param_use6.rs:14:1
    |
-LL | / fn three<T: Copy + Debug, U: Debug>(t: T, u: U) -> Two<T, U> {
-LL | |
-LL | |     (u, t)
-LL | | }
-   | |_^ expected `(T, T)`, got `(U, T)`
+LL | fn three<T: Copy + Debug, U: Debug>(t: T, u: U) -> Two<T, U> {
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `(T, T)`, got `(U, T)`
```

Comment:
This seems slightly worse than the previous message. However, I think this should really be pointing to a value being returned (either an explicit `return` or the trailing expression). This is a nightly feature, so I think it's fine to improve this later (e.g. before stabilization) as needed.

Reply:
I think we probably should do so in general, but that isn't bad enough for me to block this pr